# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/kglobalacceld/hotfix-namelink.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Upstream installs libKGlobalAccelD.so.0 and libKGlobalAccelD.so.6.7.3 only
(LIBRARY NAMELINK_SKIP), since consumers are expected to use the exported
K::KGlobalAccelD target with its absolute IMPORTED_LOCATION.

For cross builds the KDE GCC_WRAPPER_FILTER (package/kde/kcoreaddons/kde-conf.in)
rewrites absolute $root library paths back to -l<name>, which then fails:

  ld: cannot find -lKGlobalAccelD: No such file or directory
  make[2]: *** [src/CMakeFiles/kwin.dir/build.make:7041: bin/libkwin.so.6.7.3] Error 1

Install the development symlink like every other shared library so the
-l<name> form resolves.

--- kglobalacceld-6.7.3/src/CMakeLists.txt.vanilla
+++ kglobalacceld-6.7.3/src/CMakeLists.txt
@@ -72,7 +72,7 @@
 
 add_subdirectory(plugins)
 
-install(TARGETS KGlobalAccelD EXPORT KGlobalAccelDTargets ${KF_INSTALL_TARGETS_DEFAULT_ARGS} LIBRARY NAMELINK_SKIP)
+install(TARGETS KGlobalAccelD EXPORT KGlobalAccelDTargets ${KF_INSTALL_TARGETS_DEFAULT_ARGS})
 install(TARGETS kglobalacceld DESTINATION ${KDE_INSTALL_LIBEXECDIR})
 
 install(FILES
